client/resource_group/controller/controller.go

```diff
@@ -472,6 +475,43 @@ func (c *ResourceGroupsController) OnResponse(
 	return tmp.(*groupCostController).onResponse(req, resp)
 }
 
+// IsBackgroundRequest If the resource group has background jobs, we should not record consumption and wait for it.
+func (c *ResourceGroupsController) IsBackgroundRequest(ctx context.Context,
+	resourceGroupName, requestResource string) bool {
+	gc, err := c.tryGetResourceGroup(ctx, resourceGroupName)
+	if err != nil {
+		failedRequestCounter.WithLabelValues(resourceGroupName).Inc()
+		return false
+	}
+
+	gc.metaLock.RLock()
+	defer gc.metaLock.RUnlock()
+	return c.checkBackgroundSettings(ctx, gc.meta.BackgroundSettings, requestResource)
```

Comment:
```suggestion
	return c.checkBackgroundSettings(ctx, gc.getMeta().BackgroundSettings, requestResource)
```
BTW, I think it's better to replace gc.metaLock with a atomic pointer here since it only protects a single field.

Reply:
This must be a better way to do it! But I would prefer to set a `getBackgroundSettings` function, since `getMeta` is not a `Rwlock` function. What do u think? :)
```
func (gc *groupCostController) getBackgroundSettings() *rmpb.BackgroundSettings {
	gc.metaLock.RLock()
	defer gc.metaLock.RUnlock()
	return gc.meta.BackgroundSettings
}
```

Reply:
you can just replace the Lock in getMeta with RLock instead

Reply:
Okay